bagitobjecttransfer/recordtransfer/views.py

```diff
@@ -573,6 +621,11 @@ def get_submission_group(self, cleaned_form_data: dict) -> Optional[SubmissionGr
         return group
 
 
+def get_in_progress_submission(user: User, uuid: str) -> Optional[InProgressSubmission]:
+    """Retrieve an in-progress submission for a given user and UUID."""
+    return InProgressSubmission.objects.filter(user=user, uuid=uuid).first()
```

Comment:
I would prefer all functions directly defined in views.py be actual views. I recommend removing this function and extracting its functionality to the two places where it's referenced (see my other comments).

Reply:
I agree. I made some additional changes on top of this so that the InProgressSubmission object is queried only once for each instance of the TransferFormWizard during `dispatch()`. The user is also redirected to a fresh transfer form if the uuid in the query is invalid.

Resolved by:
[Refactored view to use InProgressSubmssion instance variable rather t…](https://github.com/NationalCentreTruthReconciliation/Secure-Record-Transfer/pull/240/commits/ce4b0fa60c2c9ffb80c8229683ae4799309f983b)